src/platform-implementation-js/dom-driver/gmail/views/gmail-route-view/gmail-route-view.ts

```diff
@@ -357,18 +364,70 @@ class GmailRouteView {
         eventName: 'newGmailThreadView',
         view: gmailThreadView,
       });
+    } else if (parseResult?.previewPaneContainerElement) {
+      this.#startMonitoringPreviewPaneForThread(
+        parseResult.previewPaneContainerElement,
+      );
     } else {
-      // This element is always present in thread lists, but it only has contents
-      // when in preview pane mode. We want to monitor it in either case
-      // because the user could switch into preview pane mode.
-      const previewPaneContainer = document.querySelector<HTMLElement>(
-        'div[role=main] .aia',
+      // if neither container element was found, then the page only shows thread rows
+      // (or error will be logged in #waitForThreadContainerSafe)
+    }
+  }
+
+  async #waitForThreadContainerSafe() {
+    let parseResult: {
+      threadContainerElement?: HTMLElement;
+      previewPaneContainerElement?: HTMLElement;
+    } | null = null;
+
+    try {
+      // additionally to page loading state, gmail might render content asynchronously
+      // wait until any of the content container elements appear in the DOM
+      parseResult = await waitFor(() => {
+        if (this.#destroyed) {
+          return {};
+        }
+
+        const threadContainerElement = this._getThreadContainerElement();
+        if (threadContainerElement) {
+          return { threadContainerElement };
+        }
+
+        const previewPaneContainerElement =
+          GmailElementGetter.getPreviewPaneContainerElement();
+        if (previewPaneContainerElement) {
+          return { previewPaneContainerElement };
+        }
+
+        return null;
+      }, 15_000);
```

Comment:
I wonder if we want to back off how frequently we initially check for these elements. May be related to #1088.

Reply:
hm might be related, can't find a codepath that would throw exception with selector `[role=main] .aia`